### PR TITLE
Make staging repo description configurable

### LINKED ISSUE #31: RFE: Make the description of the Nexus staging repo configurable
It would be handy to be able to provide a custom description for the staging repo instead of just a string saying which plugin created it.
Additionally, can you mention the groupId+artifactId+version in the default description of the staging repo?

### PR DESCRIPTION
**Pull request by marcphilipp**
The description can now be customized on the extension or individual
tasks and now defaults to the GAV of the root project.

In addition, the User-Agent (which is shown in Nexus UI) is now
customized to `gradle-nexus-publish-plugin/<version>`.

Resolves #31.

### CODE REVIEW

File: build.gradle.kts [hunk 1]
```diff
@@ -186,6 +189,9 @@ tasks {
         useJUnitPlatform()
         maxParallelForks = 8
     }
+    withType<Test>().matching { it.name.startsWith("compatTest") }.configureEach {
+        systemProperty("plugin.version", project.version)
```
Comment (szpak) [inline build.gradle.kts:193]: We can probably set for all the tests, not just `compatTest` (to reuse the previous section in `build.gradle`).

File: src/main/kotlin/io/github/gradlenexus/publishplugin/internal/NexusClient.kt [hunk 2]
```diff
@@ -54,6 +54,12 @@ open class NexusClient(private val baseUrl: URI, username: String?, password: St
                                 .build())
                     }
         }
+        httpClientBuilder.addInterceptor { chain ->
+            val version = javaClass.`package`.implementationVersion
```
Comment (szpak) [inline src/main/kotlin/io/github/gradlenexus/publishplugin/internal/NexusClient.kt:58]: What will be used when running from IDE? Maybe some safe fallback value `undefined` or `local` or `devel` could be explicitly used?

Comment (marcphilipp) [reply src/main/kotlin/io/github/gradlenexus/publishplugin/internal/NexusClient.kt:58]: That depends how you've set up your IDE. If test execution is delegated to Gradle, it works. If IDEA is used directly, it won't.

File: build.gradle.kts (hunk 1 above)
Comment (marcphilipp) [reply build.gradle.kts:193]: I didn't want to add an unnecessary input to the other test tasks.

File: src/main/kotlin/io/github/gradlenexus/publishplugin/internal/NexusClient.kt (hunk 2 above)
Comment (marcphilipp) [reply src/main/kotlin/io/github/gradlenexus/publishplugin/internal/NexusClient.kt]: I changed it so it works either way.

Comment (szpak) [reply src/main/kotlin/io/github/gradlenexus/publishplugin/internal/NexusClient.kt]: Thanks. I just didn't want to get NPE or something if that property is not set.

**Comment by szpak** [comment]
@marcphilipp You have some style violations.

**Pull request merged**